aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -378,12 +429,12 @@ namespace impl {
         // Decay ReturnType to ReturnType_ so that if a reference gets returned, we actually store it by value
         // and don't get a dangling reference. This is only required because some kernels still return `Tensor&`.
         using ReturnType_ = std::decay_t<typename decltype(delay_check)::template type_identity<ReturnType>>;
-        ReturnType_ output = call_functor_with_args_from_stack<KernelFunctor, AllowDeprecatedTypes>(functor_, delay_check(stack));
-        torch::jit::drop(*stack, num_inputs);
+        ReturnType_ output = call_functor_with_args_from_stack<KernelFunctor, AllowDeprecatedTypes>(functor_, dispatchKeySet, delay_check(stack));
+        drop_inputs::call<KernelFunctor>(stack);
```

Comment:
In the current implementation, both make_boxed_from_unboxed_functor and wrap_kernel_functor_unboxed need to handle the ignoring of a dispatchKeySet argument if the kernel function doesn't have any. This is, for example, why you need this drop_inputs stuff here. Could we instead implement it in only one place? For example, if kernels always get wrapped into wrap_kernel_functor_unboxed before being wrapped into make_boxed_from_unboxed_functor, then make_boxed_from_unboxed_functor would not have to handle it anymore.

Reply:
Correct me if I'm wrong: doing so would require:
- Being able to cast `DispatchKeySet` to an `ivalue`. This PR seems like an implementation detail that it'd be better to keep the jit unaware of
- We'd get a small perf penalty from having to push and pop an extra arg from the stack. This is really because we're waiting until later in the call sequence to get rid of the DispatchKeySet (and the sooner we get rid of it, the fewer function calls we have to plumb it through).

I agree that there is still a fair amount of redundancy in that file though. I'll think a little more on other ways to clean it up. The main problem I see right now is that in order to avoid passing the DispatchKeySet onto the stack, the index of each element we pull from the stack is offset by 1 compared to its corresponding index in the function schema.

Reply:
yes, we'd be betting on inlining there. It should probably work but measuring doesn't hurt.